src/ClockPanel.tsx

```diff
@@ -10,110 +10,72 @@ import './external/moment-duration-format';
 import { getTemplateSrv } from '@grafana/runtime';
 
 interface Props extends Themeable2, PanelProps<ClockOptions> {}
-interface State {
-  // eslint-disable-next-line
-  now: Moment;
-}
 
 export function getTimeZoneNames(): string[] {
   return (moment as any).tz.names();
 }
 
-class UnthemedClockPanel extends PureComponent<Props, State> {
-  timerID?: any = 0;
-  state = { now: this.getTZ(), timezone: '' };
-
-  componentDidMount() {
-    this.initTimers();
-  }
-
-  componentDidUpdate(prevProps: Props) {
-    const { options, data } = this.props;
-    const { options: prevOptions, data: prevData } = prevProps;
-
-    if (options.refresh !== prevOptions.refresh) {
-      this.initTimers();
-    }
-
-    if (prevData !== data) {
-      this.onPanelRefresh();
-    }
+function getTZ(tz?: string): Moment {
```

Comment:
Maybe call this `getMoment` or `getNowInTz` since we are not returning the time zone but a moment object potentially configured for a given time zone.

```suggestion
function getMoment(tz?: string): Moment {
```